mesonbuild/interpreter.py

```diff
@@ -3394,6 +3394,12 @@ def unpack_env_kwarg(self, kwargs) -> build.EnvironmentVariables:
         envlist = kwargs.get('env', EnvironmentVariablesHolder())
         if isinstance(envlist, EnvironmentVariablesHolder):
             env = envlist.held_object
+        elif isinstance(envlist, dict):
+            FeatureNew('environment dictionary', '0.52.0').use(self.subproject)
+            env = EnvironmentVariablesHolder()
+            for k, v in envlist.items():
+                env.set_method([k, v], {})
+            env = env.held_object
```

Comment:
I know you copied my patch for `configure_data()`, but in retrospect, I would pass the dict to `EnvironmentVariablesHolder()` constructor to avoid the duplicated code. Bonus point if you that that nice little cleanup for configure data too :-)

Reply:
Oh, and even better, move the code below that supports passing an array into `EnvironmentVariablesHolder` constructor too, so we could also support `environment(['FOO=BAR', ...])` syntax for consistency.

Reply:
ok